### PR TITLE
bigquery connection - spark connection type

### PR DESCRIPTION
**Pull request by DrFaust92**
If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
bigqueryconnection - add `spark` support
```

### CODE REVIEW

File: mmv1/products/bigqueryconnection/Connection.yaml [hunk 1]
```diff
@@ -110,6 +110,13 @@ examples:
     vars:
       connection_id: "my-connection"
       database: "projects/project/instances/instance/databases/database"
+  - !ruby/object:Provider::Terraform::Examples
+    name: "bigquery_connection_spark"
+    pull_external: true
```
Comment (ScottSuarez) [inline mmv1/products/bigqueryconnection/Connection.yaml:115]: `pull_external` shouldn't be needed. It's only needed if an external provider to the example is needed to execute the test. Like `random`

File: mmv1/products/bigqueryconnection/Connection.yaml [hunk 2]
```diff
@@ -299,10 +310,43 @@ properties:
...
+          - !ruby/object:Api::Type::String
+            name: metastoreService
+            description: Resource name of an existing Dataproc Metastore service in the form of projects/[projectId]/locations/[region]/services/[serviceId].
+      - !ruby/object:Api::Type::NestedObject
+        name: sparkHistoryServerConfig
+        description: Spark History Server configuration for the connection.
+        properties:
+          - !ruby/object:Api::Type::String
+            name: dataprocCluster
+            description: Resource name of an existing Dataproc Cluster to act as a Spark History Server for the connection if the form of projects/[projectId]/regions/[region]/clusters/[cluster_name].
```
Comment (ScottSuarez) [inline mmv1/products/bigqueryconnection/Connection.yaml:352]: These values are settable but there is no corresponding test. Could we test these sub-properties in a configuration?

**Review by ScottSuarez — CHANGES_REQUESTED** [review]

File: mmv1/products/bigqueryconnection/Connection.yaml (hunk 1 above)
Comment (DrFaust92) [reply mmv1/products/bigqueryconnection/Connection.yaml:115]: Sorry. copy pasta mistake. will remove

File: mmv1/products/bigqueryconnection/Connection.yaml (hunk 2 above)
Comment (DrFaust92) [reply mmv1/products/bigqueryconnection/Connection.yaml:352]: Sure, it will make the test at least 20 mins though. ill add those

Comment (ScottSuarez) [reply mmv1/products/bigqueryconnection/Connection.yaml:352]: mm I see. As we need to spin up a dataproc cluster for them.

Comment (DrFaust92) [reply mmv1/products/bigqueryconnection/Connection.yaml:352]: added dataproc test, i can add metastore as well but its going to take a long time to test. wdyt @ScottSuarez ?

Comment (ScottSuarez) [reply mmv1/products/bigqueryconnection/Connection.yaml:352]: let's abstain for now ! As long as you've verified locally

**Comment by ScottSuarez** [comment]
Is this not a GA feature?

```
        
        Error: Error creating Connection: googleapi: Error 400: Spark connections not supported in this project.
        
          with google_bigquery_connection.connection,
          on terraform_plugin_test.tf line 4, in resource "google_bigquery_connection" "connection":
           4: resource "google_bigquery_connection" "connection" {
```

**Review by ScottSuarez — COMMENTED** [review]
Following up on my comment about not being GA.. feel free to comment if you have any information regarding the release status or timeline of this feature. Otherwise we can hold off until the feature is released.

**Comment by ScottSuarez** [comment]
```
        Error: Error creating Connection: googleapi: Error 400: Spark connections not supported in this project.
        
          with google_bigquery_connection.connection,
          on terraform_plugin_test.tf line 4, in resource "google_bigquery_connection" "connection":
           4: resource "google_bigquery_connection" "connection" {
```

**Comment by DrFaust92** [comment]
[ScottSuarez](https://github.com/ScottSuarez) It seems to be preview, just got my hand on a project that has this. will test but ill keep it draft for now

File: mmv1/products/bigqueryconnection/Connection.yaml [hunk 3]
```diff
@@ -119,6 +119,13 @@ examples:
     vars:
       connection_id: 'my-connection'
       database: 'projects/project/instances/instance/databases/database'
+  - !ruby/object:Provider::Terraform::Examples
+    name: "bigquery_connection_spark"
+    pull_external: true
```
Comment (ScottSuarez) [inline mmv1/products/bigqueryconnection/Connection.yaml:124]:
```suggestion
```

pull_external doesn't appear to be relevant here.

**Review by ScottSuarez — CHANGES_REQUESTED** [review]
Small change requested

**Comment by DrFaust92** [comment]
Thanks [ScottSuarez](https://github.com/ScottSuarez), done

**Review by ScottSuarez — APPROVED** [review]

**Pull request merged**